bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/link/AbstractLinkRegistry.java

```diff
@@ -44,45 +123,99 @@ protected AbstractLinkRegistry(final Class<P> providerClazz) {
      * @return true if linked, false otherwise
      */
     public boolean isLinked(String itemName, UID uid) {
```

Comment:
I see you add a lot of final modifiers. Which is good for performance. You might also want to add them to the methods where they are missing from.

Reply:
> Which is good for performance.

Really? I don't think that `final` on arguments improve the performance.
It is just to be "more safe".

I use normally `final` on all places where possible if I write code and my IDE setup (non-ESH) also adds final where possible on save. I didn't touch all the existing code but I will do it if it is really performance relevant.

IIRC @kaikreuzer didn't want to introduce a guideline or formatting / save action rule that adds final on all possible places.

To be clear:
I would prefer to add final on every argument to grant they are not changed, but I assume that I should not touch all the existing code and add final (without any other change) just because of personal preference.

Reply:
Yes you are right. I also think the main performance benefit comes from declaring local variables as final. @amitjoy wrote some [articles](https://dzone.com/articles/all-about-finals) about it. Since the link registry code is called so often, it may be worth it to use final local variables in this code where ever possible.

Reply:
There is some additional difference:
The example in the article is using final for values that are already known at compile time and so the byte code could already be optimized.
I assume the example will look different if the value is not known at compile time but the string is chosen at runtime (but I could be wrong).

Reply:
... but I added final where possible